tensorflow/python/kernel_tests/nn_ops/conv_ops_test.py

```diff
@@ -1154,6 +1157,10 @@ def _RunAndVerifyBackpropFilter(self,
       if isinstance(padding, (list, tuple)):
         new_padding = test_util.NHWCToNCHW(new_padding)
     for dtype in self._DtypesToTest(use_gpu=use_gpu):
+      # The backprop_filter may generate high err with a sequence of integers as
+      # the input (See _CreateNumpyTensor). So, we skip this dtype.
+      if dtype == dtypes.bfloat16:
```

Comment:
We don't have any other backprop filter tests, which is problematic. Maybe it will pass if you round `expected` to the nearest bfloat16 value? If that still fails, maybe adding an `err_bf16` argument? (Note `assertAllCloseAccordingToType` takes a separate `bfloat16_rtol` arg.) Alternatively you can add a `skip_bf16` argument so you can disable only the failing tests.

Reply:
Thanks for the feedback. I can try to check how far away the results are from the `expected` and play with the error/tolerance. Btw, can you point me to an example of `skip_bf16`? How such arg works?

Reply:
I would start by rounding `expected` to the nearest bf16 value, by doing something like `expected = np.array(expected, dtype=tf.bfloat16.as_numpy_dtype)`. That might reduce the tolerance you need.

There is no existing example of `skip_bf16`. Basically, the idea is you add a `skip_bf16` argument to `_RunAndVerifyBackpropFilter` with a default value is False. Then, at this point in the code on line 1162, you do:

```
if dtype == dtypes.bfloat16 and skip_bf16:
  continue
```

That way, if the caller passes `skip_bf16=True`, the test is not run with bf16. Then, for tests that call `_RunAndVerifyBackpropFilter` but fail with bf16, you can pass `skip_bf16=True` to make the test pass.

Reply:
Thanks for the suggestion. I just reran the tests and all of them are passed this time. So, I think the skip_bf16 is no longer needed. Removing this check.